### ISSUE TITLE
Can’t read NE time zones shapefile?

### ISSUE THREAD

**Issue by mbostock** [mbostock/shapefile#36]
[ne_10m_time_zones.zip](https://github.com/mbostock/shapefile/files/1096332/ne_10m_time_zones.zip)

**Comment by mbostock** [comment]
Steps to reproduce:

```
$ curl -o ne_10m_time_zones.zip http://naciscdn.org/naturalearth/10m/cultural/ne_10m_time_zones.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1923k  100 1923k    0     0   173k      0  0:00:11  0:00:11 --:--:--  175k
```

```
$ unzip ne_10m_time_zones.zip
Archive:  ne_10m_time_zones.zip
  inflating: ne_10m_time_zones.README.html  
 extracting: ne_10m_time_zones.VERSION.txt  
  inflating: ne_10m_time_zones.dbf   
  inflating: ne_10m_time_zones.prj   
  inflating: ne_10m_time_zones.shp   
  inflating: ne_10m_time_zones.shx   
```

```
$ shp2json ne_10m_time_zones.shp -o ne_10m_time_zones.json

  error: Invalid array length

```

**Comment by mbloch** [comment]
Hi Mike,
You might also take a look at ne_10m_railroads.shp as a test case (I have N.E. version 2.0.0), which has 172 garbage bytes between many of the records, a few records with duplicate record ids. In order to handle this file, I wrote code to probe ahead in the .shp file in 4-byte increments.

**Comment by mbloch** [comment]
[railroads.zip](https://github.com/mbostock/shapefile/files/1186017/railroads.zip)

**Comment by mbostock** [comment]
Thanks for the extra test case! It seems I still have some work to-do in #37!

**Comment by mbloch** [comment]
Sure thing! Because of files like these, I considered using the .shx index file to help with parsing .shp files... they are so rare, though, I decided just to keep using just the .shp file and probe for the next record when there are garbage bytes.

**Comment by mbostock** [comment]
Made some minor changes in 74dfc83907eb185d28c154fc237f273aa0d3b578 and seems fine now.

**Comment by mbostock** [comment]
Fixed in 0.6.3!
